playbooks/common/openshift-cluster/upgrades/v3_0_to_v3_1/upgrade.yml

```diff
@@ -235,6 +238,11 @@
 
   - name: Upgrade master packages
     command: "{{ ansible_pkg_mgr}} update -y {{ openshift.common.service_type }}-master{{ openshift_version }}"
+    when: openshift_pkg_version is not defined
+
+  - name: Upgrade packages
+    command: "{{ ansible_pkg_mgr}} install -y {{ openshift.common.installed_product_rpms | oo_31_rpm_rename_conversion(openshift_version) | join (' ')}}"
```

Comment:
@brenton it's because it's a different play...  openshift_facts is called for the play that starts at 221 and ends at 229, 231 starts a separate play, but it doesn't look like it has to.